tests/src/test/java/alluxio/client/cli/fs/command/LoadMetadataCommandV2IntegrationTest.java

```diff
@@ -0,0 +1,295 @@
+    assertTrue(mOutput.toString().contains("State: SUCCEEDED"));
+  }
+
+  // The main idea of this test is start an async loadMetadata task and cancel it when it's running
+  // I think this is difficult to fix...
+  @Test
+  public void loadMetadataTestV2cancel() {
+    for (int i = 0; i < 1; i++) {
+      mS3Client.putObject(TEST_BUCKET, TEST_FILE + i, TEST_CONTENT);
+    }
+    mOutput.reset();
+    AlluxioURI uriDir = new AlluxioURI("/" );
+    CompletableFuture<Void> future = CompletableFuture.supplyAsync(() -> {
+      mFsShell.run("loadMetadata", "-v2", "-R", "-a", uriDir.toString());
+      return null;
+    });
+    String id;
+    Pattern pattern = Pattern.compile("Task group (\\d+)");
+    Matcher matcher = pattern.matcher(mOutput.toString());
+    while(!matcher.find()) {
+      matcher = pattern.matcher(mOutput.toString());
+    }
+    id = matcher.group(1);
+    FileSystemShell anotherFsShell = new FileSystemShell(Configuration.global());
+
+    // Cancel a running task
+    anotherFsShell.run("loadMetadata", "-v2", "-o", "cancel", "-id", id);
+    assertTrue(mOutput.toString().contains(String.format("Task group %s cancelled", id)));
+
+    // Cancel a canceled task
+    mOutput.reset();
+    anotherFsShell.run("loadMetadata", "-v2", "-o", "cancel", "-id", id);
+    assertTrue(mOutput.toString().contains(String.format("Task %s not found or has already been canceled", id)));
+
+    // Trying to cancel completed task
+    mOutput.reset();
+    mFsShell.run("loadMetadata", "-v2", "-R", "-a", uriDir.toString());
+    matcher = pattern.matcher(mOutput.toString());
+    while(!matcher.find()) {
+      matcher = pattern.matcher(mOutput.toString());
+    }
+    id = matcher.group(1);
+    mFsShell.run("loadMetadata", "-v2", "-o", "cancel", "-id", id);
+    assertTrue(mOutput.toString().contains(String.format("Task %s not found or has already been canceled", id)));
+  }
+
+  @Test
+  public void loadMetadataTestV2R() throws IOException, AlluxioException {
+    // dir number
+    int dirCount = 3;
+    // Child number for one dir
+    int fileCount = 10;
+    for (int dirIndex = 0; dirIndex < dirCount; dirIndex++) {
+      for (int fileIndex = 0; fileIndex < fileCount; fileIndex++) {
+        mS3Client.putObject(TEST_BUCKET, "test" + dirIndex + "/" + fileIndex, TEST_CONTENT);
+      }
+    }
+    AlluxioURI uriDir = new AlluxioURI("/" );
+
+    mFsShell.run("loadMetadata", "-v2", "-a", uriDir.toString());
+    assertTrue(mOutput.toString().contains("State: SUCCEEDED"));
+    GetStatusPOptions getStatusPOptions =
+        GetStatusPOptions.newBuilder().setLoadMetadataType(LoadMetadataPType.NEVER).build();
+    URIStatus statusAfter = null;
+    for (int i = 0; i < dirCount; i++) {
+      statusAfter = mFileSystem.getStatus(new AlluxioURI("/test" + i), getStatusPOptions);
+      assertEquals(statusAfter.getFileInfo().getLength(), 0);
+    }
+    // avoid legacy "State: SUCCEEDED"
```

Comment:
what does avoid legacy mean?

Reply:
To avoid match previous target string in later validation. There is 'State: SUCCEEDED' in mOutput from previous loadMetadata operation

Reply:
remove this comment... not very informative.